plugins/modules/zpool.py

```diff
@@ -0,0 +1,654 @@
+            if not saw_pool:
+                if device == self.name:
+                    saw_pool = True
+                continue
+            if re.match(r'^(mirror|raidz\d?)\-\d+$', device) or device in ('cache', 'logs', 'spares'):
+                if current:
+                    vdevs.append(current)
+                vdev_type = ('stripe' if device in ('cache', 'logs', 'spares') else ('mirror' if device.startswith('mirror') else 'raidz'))
+                current = {'name': device, 'type': vdev_type, 'disks': []}
+                continue
+            if device.startswith('/') and current:
+                current['disks'].append(self.base_device(device))
+                continue
+            if device.startswith('/'):
+                base_device = self.base_device(device)
+                vdevs.append({'name': base_device, 'type': 'stripe', 'disks': [base_device]})
+        if current:
+            vdevs.append(current)
+        return vdevs
+
+    def remove_vdevs(self):
+        current = self.list_vdevs_with_names()
+        current_disks = {disk for vdev in current for disk in vdev['disks']}
+        desired_disks = {disk for vdev in self.vdevs for disk in vdev.get('disks', [])}
+        gone = current_disks - desired_disks
+        to_remove = [vdev['name'] for vdev in current if any(disk in gone for disk in vdev['disks'])]
+        if not to_remove:
+            return {}
+        with self.zpool_runner('subcommand dry_run name vdev_name', check_rc=True) as ctx:
+            rc, stdout, stderr = ctx.run(
+                subcommand='remove',
+                dry_run=self.module.check_mode,
+                name=self.name,
+                vdev_name=to_remove,
+            )
+        self.changed = True
+        if self.module.check_mode:
+            return {'prepared': stdout}
+        before = [vdev['name'] for vdev in current]
+        after = [name for name in before if name not in to_remove]
+        return {'before': {'vdevs': before}, 'after': {'vdevs': after}}
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            name=dict(type='str', required=True),
+            state=dict(type='str', choices=['present', 'absent'], default='present'),
+            disable_new_features=dict(type='bool', default=False),
+            force=dict(type='bool', default=False),
+            pool_properties=dict(type='dict', default={}),
+            filesystem_properties=dict(type='dict', default={}),
+            mountpoint=dict(type='str', required=False),
+            altroot=dict(type='str', required=False),
+            temp_name=dict(type='str', required=False),
+            vdevs=dict(
+                type='list',
+                elements='dict',
+                required=False,
+                options=dict(
+                    role=dict(
+                        type='str',
+                        choices=['log', 'cache', 'spare', 'dedup', 'special'],
+                        required=False,
```

Comment:
`required=False` is redundant in the arg_spec, please remove it here and in the other options as well